server/webapp/WEB-INF/rails/webpack/views/components/buttons/index.tsx

```diff
@@ -30,6 +30,11 @@ export enum ButtonIcon {
   FILTER,
 }
 
+export enum Alignment {
```

Comment:
@kierarad a simpler alternative to using a TypeScript `enum` is to just do:

```typescript
export type Alignment = "right" | "left";
```

Then, you can just pass in the string literals `"right"` and `"left"` to the `vnode.attrs` object instead of `Alignment.right`, etc. TypeScript will still validate (and, in fact, fail to compile) if the values are not one of the options.

This becomes:

```jsx
<Buttons.Primary align="right" ... />
```

And the `alignClass()` implementation becomes:

```typescript
private static alignClass(align?: Alignment, css?: Styles) {
  css = css || defaultStyles;

  switch (align) {
    case "left":
      return css.leftAlign;
    case "right":
      return css.rightAlign;
  }
  return "";
}
```

Reply:
@marques-work thanks! I actually prefer this as well. I'll change

Reply:
@kierarad cool :)